test/integration/test_konnect_entities.go

```diff
@@ -0,0 +1,124 @@
+package integration
+
+import (
+	"testing"
+	"time"
+
+	"github.com/Kong/sdk-konnect-go/models/components"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	testutils "github.com/kong/gateway-operator/pkg/utils/test"
+	"github.com/kong/gateway-operator/test"
+	"github.com/kong/gateway-operator/test/helpers"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+)
+
+func TestKonnectEntities(t *testing.T) {
+	ns, _ := helpers.SetupTestEnv(t, GetCtx(), GetEnv())
```

Comment:
Why don't we need a `cleaner` to be created here and all later-created objects added to it?

Reply:
A cleaner is created underneath anyway, and a whole namespace is deleted eventually. I couldn't use a cleaner to delete objects because it handles deletes in FIFO order and that won't work in this case: `KonnectAPIAuthConfiguration` shouldn't be deleted before any other object as that is required for others to complete their finalizer which is deleting a reflecting entity in Konnect. That's why I'm only cleaning up a `KonnectControlPlane` and waiting for its deletion synchronously to ensure it was successfully deleted along with its children, and the `KonnectAPIAuthConfiguration` is implicitly deleted with the namespace.